### ISSUE TITLE
Argument error: expected 64-byte String, got 3

### ISSUE THREAD

**Issue by ekampp** [net-ssh/net-ssh#638]
### Expected behavior

I'm able to connect to the server

I have recently begun upgrading my ssh keys to the new ECDSA, and I got a helpful message about adding the `ed25519` and `bcrypt_pbkdf` gems which I did. After this I reached this error, which don't seem to be super helpful.

### Actual behavior

Generating a key with `ssh-keygen -b 4096 -C "<USER>@<IP>"`

I get an `ArgumentError` with: "expected 64-byte String, got 3"

If I generate the key specifically with `-t ecdsa -b 512`

I get "expected 64-byte String, got 133"

### System configuration

- net-ssh version: 5.0.2
- Ruby version: 2.5.1p57

### Example App

```
require 'net/ssh'

opts = {
  :host => '<IP>',
  :port => 22,
  :user => '<USER>'
}

opts[:logger] = Logger.new(STDERR)
opts[:logger].level = Logger::INFO

Net::SSH.start(opts[:host], opts[:user], :keys_only => true, :logger => opts[:logger]) do |ssh|
  ls_result = ssh.exec("ls -l")
  puts ls_result
end
```

**Comment by stiller-leser** [comment]
Not a solution to this issue, but a workaround from https://github.com/duplicati/duplicati/issues/3360#issuecomment-418826330. Unfortunately this requires you to generate a new key in the old key format using ` ssh-keygen -m PEM -t rsa -b 4096`. The reasoning for the switch to the new format can be found here: https://https.www.google.com.tedunangst.com/flak/post/new-openssh-key-format-and-bcrypt-pbkdf

**Comment by SharkFourSix** [comment]
> Not a solution to this issue, but a workaround from [duplicati/duplicati#3360 (comment)](https://github.com/duplicati/duplicati/issues/3360#issuecomment-418826330). Unfortunately this requires you to generate a new key in the old key format using ` ssh-keygen -m PEM -t rsa -b 4096`. The differences are described here: https://https.www.google.com.tedunangst.com/flak/post/new-openssh-key-format-and-bcrypt-pbkdf
> 
> There is also a discussion about it here #633, with the bug apparently located here: https://github.com/net-ssh/net-ssh/blob/master/lib/net/ssh/key_factory.rb#L112

Sadly this does not seem to work.

**Comment by mfazekas** [comment]
@SharkFourSix this should be fixed in 5.2.0, isn't it?

**Comment by quickdudley** [comment]
@mfazekas An issue producing the same error text was fixed between net-ssh 5.0.2 and 5.2.0. But since the workaround described by @SharkFourSix wasn't working for me I can't confirm that it was the same issue.
